### DISCUSSION TITLE
How to install only a single Boost feature in vcpkg manifest mode

### DISCUSSION THREAD

**Discussion by BullyWiiPlaza** [microsoft/vcpkg#31760] [Q&A] [answered]
Let's say I want to use the Boost header `#include <boost/algorithm/string/replace.hpp>` in my project and I'm writing a `vcpkg.json` file (manifest file) for it. Obviously I don't want to include the entirety of Boost for a simple algorithm header so I'd prefer to just use the algorithm module. If I specify `boost-algorithm` in my `dependencies` node, I still get an installation of more or less the entirety of Boost. Using features via e.g.

```
{
	"name": "boost",
	"features": [
		"algorithm"
	]
}
```

does not work, it will yield the error message `1>C:\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets(183,5): warning : error: boost:x64-windows@1.82.0#1 does not have required feature algorithm`. Isn't the algorithm header more or less standalone? Am I doing something wrong or is it not possible to quickly install just `boost-algorithm`?

**Comment by autoantwort** [comment]
Seems that `boost-algorithm` simply has these dependencies: 
![output1](https://github.com/microsoft/vcpkg/assets/41973254/15c5a036-3699-4506-9529-9ef3ead1f4fb)

**Reply by autoantwort** [reply, marked as answer]
You "only" install 39 out of 147 boost packages.

**Reply by autoantwort** [reply]
And here the complete boost dependency graph :D 
![output1](https://github.com/microsoft/vcpkg/assets/41973254/03d4047a-0e59-4a67-8a09-713b18de2492)

**Reply by Neumann-A** [reply]
> And here the complete boost dependency graph :D

I think we all agree that this is a mess ;)

**Reply by Osyotr** [reply]
Note that boosts ports dependencies are generated heuristically so it's entirely possible that some of them are redundant (example: https://github.com/microsoft/vcpkg/pull/20575)
